easybuild/tools/containers/singularity.py

```diff
@@ -98,8 +113,15 @@
 eval "$@"
 
 %%environment
+# make sure that 'module' and 'ml' commands are defined
 source /etc/profile
+# increase threshold time for Lmod to write cache in $HOME (which we don't want to do)
+export LMOD_SHORT_TIME=86400
+# avoid picking up modules from outside of container
+module unuse $MODULEPATH
```

Comment:
the environment is passed right? So what happens when a module is loaded when you start the container? Lmod will see it inside the container too?

Reply:
Yes, indeed.

```
[kehoste@eb-ci ~]$ ml gompi/2018b
[kehoste@eb-ci ~]$ ml

Currently Loaded Modules:
  1) GCCcore/7.3.0                 3) GCC/7.3.0-2.30              5) numactl/2.0.11-GCCcore-7.3.0   7) libxml2/2.9.8-GCCcore-7.3.0       9) hwloc/1.11.10-GCCcore-7.3.0   11) gompi/2018b
  2) binutils/2.30-GCCcore-7.3.0   4) zlib/1.2.11-GCCcore-7.3.0   6) XZ/5.2.4-GCCcore-7.3.0         8) libpciaccess/0.14-GCCcore-7.3.0  10) OpenMPI/3.1.1-GCC-7.3.0-2.30



[kehoste@eb-ci ~]$ singularity shell test.sif

Inactive Modules:
  1) GCC/7.3.0-2.30     3) OpenMPI/3.1.1-GCC-7.3.0-2.30     5) binutils/2.30-GCCcore-7.3.0     7) hwloc/1.11.10-GCCcore-7.3.0         9) libxml2/2.9.8-GCCcore-7.3.0     11) zlib/1.2.11-GCCcore-7.3.0
  2) GCCcore/7.3.0      4) XZ/5.2.4-GCCcore-7.3.0           6) gompi/2018b                     8) libpciaccess/0.14-GCCcore-7.3.0    10) numactl/2.0.11-GCCcore-7.3.0

Singularity test.sif:~> ml

Currently Loaded Modules:
  1) bzip2/1.0.6

Inactive Modules:
  1) GCCcore/7.3.0                 3) GCC/7.3.0-2.30              5) numactl/2.0.11-GCCcore-7.3.0   7) libxml2/2.9.8-GCCcore-7.3.0       9)  hwloc/1.11.10-GCCcore-7.3.0    11) gompi/2018b
  2) binutils/2.30-GCCcore-7.3.0   4) zlib/1.2.11-GCCcore-7.3.0   6) XZ/5.2.4-GCCcore-7.3.0         8) libpciaccess/0.14-GCCcore-7.3.0   10) OpenMPI/3.1.1-GCC-7.3.0-2.30
```

Do you think we should do a (forced) `ml purge` first?